server/controllers/userController.js

```diff
@@ -72,11 +72,12 @@ const updateProfile = async (req, res) => {
                         return res.status(404).json({ error: 'User not found' });
                 }
 
-                // Update user's profile with provided fields or defaults
-                user.username = username || 'Anonymous';
-                user.aboutMe = aboutMe || null;
-                user.gender = gender || 'Unknown';
-                user.age = age || null;
+                // Update user's profile with provided fields or the User fields or defaults
+                user.username = username || user.username || 'Anonymous';
+                user.aboutMe = aboutMe || user.aboutMe || null;
+                user.gender = gender || user.gender || 'Unknown';
+                user.age = age || user.age || null;
```

Comment:
next time if you want to make a change that isn't part of your PR, create a new issue, will let this slide for now.

Reply:
Sure, will keep that in mind.